### DISCUSSION TITLE
Is there a way to Integrate clients with Pihole with DNS?

### DISCUSSION THREAD

**Discussion by mygitcrazy** [netbirdio/netbird#1369] [Issue Triage]
Is there a way to use a pihole dns for the connected clients that use the same dns as Pihole.

So that connected clients are filtered though Pihole....
How would i achieve this?

I'm thinking about something like this approach?
https://tailscale.com/kb/1114/pi-hole/
But I'm not sure if that fits with netbird

**Comment by Darin755** [comment]
I think netbord has a DNS feature that allows you to set custom DNS. I'm not sure if it will work for what your doing but it might.

Another option is to point your DNS to your pihole and then have a secondary DNS for when netbird starts up. I'm not experienced with DNS or pihole so this may not work.

**Reply by mygitcrazy** [reply]
Sorry for the late reply...
But thanks anyway for the response 👍 

I'm going to deepdive more into this later on...

**Comment by buzzzo** [comment]
Pihome is just a dns server with filtering abilitiers.
Just create a nameserver on the netbird ui and pass your pihole ip as dns server

**Reply by maricn** [reply]
plus, you might want to add `pihole-FTL.service` as a precondition for the `netbird.service`, otherwise it can hijack `:53` and make pihole service unable to start..
